inference_perf/client/metricsclient/prometheus_client/google_managed_prometheus_client.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2025 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+from pydantic import HttpUrl
+import requests
+from inference_perf.client.metricsclient.prometheus_client.base import PrometheusMetricsClient
+from inference_perf.config import PrometheusClientConfig
+import google.auth
+
+logger = logging.getLogger(__name__)
+
+
+class GoogleManagedPrometheusMetricsClient(PrometheusMetricsClient):
+    def __init__(self, config: PrometheusClientConfig) -> None:
+        if config.google_managed is None:
+            raise Exception("google managed prometheus config missing")
+        self.google_managed_prometheus_config = config.google_managed
+        # Creates a credentials object from the default service account file
+        # Assumes that script has appropriate default credentials set up, ref:
+        # https://googleapis.dev/python/google-auth/latest/user-guide.html#application-default-credentials
+        credentials, project_id = google.auth.default()  # type: ignore[no-untyped-call]
+        self.credentials = credentials
+        self.project_id = project_id
+        config.url = HttpUrl(f"https://monitoring.googleapis.com/v1/projects/{self.project_id}/location/global/prometheus")
+        super().__init__(config)
+
+    def execute_query(self, query: str, eval_time: str) -> float:
+        """
+        Executes the given query on the Prometheus server and returns the result.
+
+        Args:
+        query: the PromQL query to execute
+        eval_time: the time at which the query is evaluated, used to ensure we are querying the correct time range
+
+        Returns:
+        The result of the query.
+        """
+        query_result = 0.0
+
+        # Prepare an authentication request - helps format the request auth token
+        auth_req = google.auth.transport.requests.Request()
+
+        self.credentials.refresh(auth_req)
+        headers = {"Authorization": "Bearer " + self.credentials.token}
```

Comment:
Do we need a reimplementation of the entire execute_query function with GMP or only authentication and passing in the bearer token is enough? If there are no other differences, it might make sense to avoid duplicating the rest of the logic here.